src/Dirstructure.cpp

```diff
@@ -239,6 +242,65 @@ wxString Dirstructure::MaximaDefaultLocation()
   return wxT("maxima");
 }
 
+wxString Dirstructure::GnuplotDefaultLocation(wxString pathguess)
+{
+  wxString gnuplot_binary;
+
+  if(wxFileName(pathguess).IsAbsolute())
+    gnuplot_binary = pathguess;
+  else
+  {
+    wxPathList pathlist;
+
+    // Add paths relative to the path of the wxMaxima executable
+    pathlist.Add(wxFileName(wxStandardPaths::Get().GetExecutablePath()).GetPath());
+    pathlist.Add(wxFileName(wxStandardPaths::Get().GetExecutablePath()).GetPath()+"/../");
+    pathlist.Add(wxFileName(wxStandardPaths::Get().GetExecutablePath()).GetPath()+"/../gnuplot");
+    pathlist.Add(wxFileName(wxStandardPaths::Get().GetExecutablePath()).GetPath()+"/../gnuplot/bin");
+
+    // Add paths from the PATH environment variable
+    pathlist.AddEnvList(wxT("PATH"));
+
+    // Add OSX specific paths
+#ifdef __WXOSX__
+    // MacPorts:
+    // The MacPorts default binary path /opt/local/bin/ is not in the PATH for applications.
+    // It is added to .profile, but this is only used by shells.
+    // => Add the default MacPorts binary path /opt/local/bin/ to our search path list.
+    //
+    // Homebrew:
+    // Homebrew installs binaries in /usr/local/bin, which is in the PATH by default.
+    //
+    // Application packages including gnuplot:
```

Comment:
If you want to add more wxLogMessages somewhere here: Feel free to do so. Most messages originally were meant as pure eye-candy. But if anything goes completely wrong adding *loads* of log messages has proven helpful for finding out where a problem might reside on a system that doesn't contain a debugger and misbehaves.